### ISSUE TITLE
Master fails to load properties file that can be read by stable

### ISSUE THREAD

**Issue by bethac07** [CellProfiler/CellProfiler-Analyst#224]
Running CPA from source on a properties file that is using per-image classification, I get the error below.  It runs fine if I check out 2.2.1 rather than the master.  Posted just the top and bottom of the properties file but I can post any other relevant parts you might want.

```
WARNING:root:PROPERTIES WARNING (channels_per_image): No value(s) specified. CPA will assume 1 channel per image.
INFO:root:PROPERTIES: Using default image_buffer_size=1
INFO:root:PROPERTIES: Using default tile_buffer_size=1
INFO:root:PROPERTIES: Per-Object classes will be written to table "AstrocyteQC"
WARNING:root:PROPERTIES WARNING (well_format): Field was not defined, using default format of "A01".
INFO:root:[MainThread] Connecting to the database...
INFO:root:[MainThread] SQLite file: \\argon\imaging_analysis\redacted\AstrocyteQC_fixed.db
Traceback (most recent call last):
  File "CellProfiler-Analyst.py", line 552, in <module>
    app.Start()
  File "CellProfiler-Analyst.py", line 463, in Start
    db.connect()
  File "I:\GitHub\CellProfiler-Analyst\cpa\dbconnect.py", line 453, in connect
    self.CreateObjectCheckedTable()
  File "I:\GitHub\CellProfiler-Analyst\cpa\dbconnect.py", line 1469, in CreateObjectCheckedTable
    all_cols = [str(x) for x in self.GetColumnNames(p.object_table[:-8])]
  File "I:\GitHub\CellProfiler-Analyst\cpa\dbconnect.py", line 811, in GetColumnNames
    self.execute('SELECT * FROM %s LIMIT 1'%(table))
  File "I:\GitHub\CellProfiler-Analyst\cpa\dbconnect.py", line 64, in fn
    return f(db, *args, **kwargs)
  File "I:\GitHub\CellProfiler-Analyst\cpa\dbconnect.py", line 529, in execute
    '\nSecond exception was: %s'%(connID, query, e, e2))
cpa.dbconnect.DBException: ERROR: Database query failed for connection "MainThread" and failed to reconnect
Query was: "SELECT * FROM AstrocyteQC_Per_Image_ObjectTable LIMIT 1"
First exception was: no such table: AstrocyteQC_Per_Image_ObjectTable
Second exception was: ERROR: Database query failed for connection "MainThread"
Query was: "SELECT * FROM AstrocyteQC_Per_Image_ObjectTable LIMIT 1"
Exception was: no such table: AstrocyteQC_Per_Image_ObjectTable
```

```
#Thu Jun 30 19:40:46 2016
# ==============================================
#
# CellProfiler Analyst 2.0 properties file
#
# ==============================================

# ==== Database Info ====
db_type         = sqlite
db_sqlite_file  = \\argon\imaging_analysis\redacted\AstrocyteQC_fixed.db

# ==== Database Tables ====
image_table   = AstrocyteQC_Per_Image 
object_table  =

# ==== Database Columns ====
# Specify the database column names that contain unique IDs for images and
# objects (and optionally tables).
#
# table_id (OPTIONAL): This field lets Classifier handle multiple tables if
#          you merge them into one and add a table_number column as a foreign
#          key to your per-image and per-object tables.
# image_id: must be a foreign key column between your per-image and per-object
#           tables
# object_id: the object key column from your per-object table

image_id      = ImageNumber
object_id     =
plate_id      = Image_Metadata_Plate
well_id       = Image_Metadata_Well


# ======== Classification type ========
# OPTIONAL
# CPA 2.2.0 allows image classification instead of object classification.
# If left blank or set to "object", then Classifier will fetch objects (default).
# If set to "image", then Classifier will fetch whole images instead of objects.

classification_type  = image

# ======== Output Per-Object Classes ========
# OPTIONAL
# Here you can specify a MySQL table in your Database where you would like
# Classifier to write out class information for each object in the
# object_table

class_table  = AstrocyteQC

# ======== Check Tables ========
# OPTIONAL
# [yes/no]  You can ask classifier to check your tables for anomalies such
# as orphaned objects or missing column indices.  Default is on.
# This check is run when Classifier starts and may take up to a minute if
# your object_table is extremely large.

check_tables = yes
```

**Comment by daviddao** [comment]
Strange. @jhung0 any ideas?

**Comment by braymp** [comment]
Try setting <code>check_tables</code> to 'no'

**Comment by bethac07** [comment]
The workaround works!  Still probably worth investigating what change killed it though.

**Comment by braymp** [comment]
Definitely. I've been having problems with the table checking for some time now, so I set it to 'no' at the first.

**Comment by jhung0** [comment]
I'll investigate, but for now, I'll just set the default to no

**Comment by jhung0** [comment]
Ok I didn't test table checking on per image classification, so that's why the problem appears.

**Comment by jhung0** [comment]
@braymp Problems with table checking for image classification or object classification too?

**Comment by braymp** [comment]
An example: If I load the examples.properties in cpa_example fresh (i.e, the SQLite has not been created yet) with <code>check_tables = yes</code> and with object classification (i.e, no 'classification_type' field), I get this error:
```
Traceback (most recent call last):
  File "CellProfiler-Analyst.py", line 552, in <module>
    app.Start()
  File "CellProfiler-Analyst.py", line 463, in Start
    db.connect()
  File "c:\Users\brayma2\Desktop\github\cellprofiler-analyst\cpa\dbconnect.py",
line 443, in connect
    raise e
sqlite3.OperationalError: no such table: per_object_Checked
```

**Comment by braymp** [comment]
Furthermore, let's say I change the properties file to have <code>check_tables = yes</code> and enable object classification. Then Classifier loads fine with the example training set.

But if I follow this by:
* Closing CPA
* Editing the properties file to still have <code>check_tables = yes</code> but <code>classification_type = image</code>
* Adding the appropriate <code>image_width</code> and <code>image_width</code> fields, both set to 512.

I then get this error:
```
Traceback (most recent call last):
  File "CellProfiler-Analyst.py", line 552, in <module>
    app.Start()
  File "CellProfiler-Analyst.py", line 463, in Start
    db.connect()
  File "c:\Users\brayma2\Desktop\github\cellprofiler-analyst\cpa\dbconnect.py", line 453, in connect
    self.CreateObjectCheckedTable()
  File "c:\Users\brayma2\Desktop\github\cellprofiler-analyst\cpa\dbconnect.py", line 1469, in CreateObjectCheckedTable
    all_cols = [str(x) for x in self.GetColumnNames(p.object_table[:-8])]
  File "c:\Users\brayma2\Desktop\github\cellprofiler-analyst\cpa\dbconnect.py", line 811, in GetColumnNames
    self.execute('SELECT * FROM %s LIMIT 1'%(table))
  File "c:\Users\brayma2\Desktop\github\cellprofiler-analyst\cpa\dbconnect.py", line 64, in fn
    return f(db, *args, **kwargs)
  File "c:\Users\brayma2\Desktop\github\cellprofiler-analyst\cpa\dbconnect.py", line 529, in execute
    '\nSecond exception was: %s'%(connID, query, e, e2))
cpa.dbconnect.DBException: ERROR: Database query failed for connection "MainThread" and failed to reconnect
Query was: "SELECT * FROM per_image_ObjectTable LIMIT 1"
First exception was: no such table: per_image_ObjectTable
Second exception was: ERROR: Database query failed for connection "MainThread"
Query was: "SELECT * FROM per_image_ObjectTable LIMIT 1"
Exception was: no such table: per_image_ObjectTable
```

**Comment by jhung0** [comment]
First problem should be fixed by d5c0ed0

**Comment by jhung0** [comment]
Second problem should be fixed by fb2d663

**Comment by braymp** [comment]
The first issue appears to be fixed. Trying the 2nd now produces this error:
```
Traceback (most recent call last):
  File "CellProfiler-Analyst.py", line 552, in <module>
    app.Start()
  File "CellProfiler-Analyst.py", line 463, in Start
    db.connect()
  File "c:\Users\brayma2\Desktop\github\cellprofiler-analyst\cpa\dbconnect.py",line 453, in connect
    self.CreateObjectCheckedTable()
  File "c:\Users\brayma2\Desktop\github\cellprofiler-analyst\cpa\dbconnect.py",line 1478, in CreateObjectCheckedTable
    all_cols = [str(x) for x in self.GetColumnNames(object_table)]
  File "c:\Users\brayma2\Desktop\github\cellprofiler-analyst\cpa\dbconnect.py", line 811, in GetColumnNames
    self.execute('SELECT * FROM %s LIMIT 1'%(table))
  File "c:\Users\brayma2\Desktop\github\cellprofiler-analyst\cpa\dbconnect.py",line 64, in fn
    return f(db, *args, **kwargs)
  File "c:\Users\brayma2\Desktop\github\cellprofiler-analyst\cpa\dbconnect.py",line 529, in execute
    '\nSecond exception was: %s'%(connID, query, e, e2))
cpa.dbconnect.DBException: ERROR: Database query failed for connection "MainThread" and failed to reconnect
Query was: "SELECT * FROM per_image_objecttable LIMIT 1"
First exception was: no such table: per_image_objecttable
Second exception was: ERROR: Database query failed for connection "MainThread"Query was: "SELECT * FROM per_image_objecttable LIMIT 1"
Exception was: no such table: per_image_objecttable
```
Looks like the table name was made lower-case when it shouldn't be?

**Comment by jhung0** [comment]
It should be lowercase. Can you look at your database and see if any new table is created?

**Comment by bethac07** [comment]
Just got hit by this (exception below) existing SQLite was for per image classification, had Batch1_QC_Per_Image plus 2 link tables- now also has Batch1_QC_Per_Image_objecttable_checked

```
INFO:root:PROPERTIES: Per-Object classes will be written to table "BatchQC"
INFO:root:[MainThread] Connecting to the database...
INFO:root:[MainThread] SQLite file: C:\Users\[REDACTED]\20170123.db
INFO:root:Batch1_QC_Per_Image_objecttable_checked table added to database
Batch1_QC_Per_Image_objecttable_checked
Batch1_QC_Per_Image_objecttable
Traceback (most recent call last):
  File ".\CellProfiler-Analyst.py", line 552, in <module>
    app.Start()
  File ".\CellProfiler-Analyst.py", line 463, in Start
    db.connect()
  File "I:\GitHub\CellProfiler-Analyst\cpa\dbconnect.py", line 453, in connect
    self.CreateObjectCheckedTable()
  File "I:\GitHub\CellProfiler-Analyst\cpa\dbconnect.py", line 1478, in CreateObjectCheckedTable
    all_cols = [str(x) for x in self.GetColumnNames(object_table)]
  File "I:\GitHub\CellProfiler-Analyst\cpa\dbconnect.py", line 811, in GetColumnNames
    self.execute('SELECT * FROM %s LIMIT 1'%(table))
  File "I:\GitHub\CellProfiler-Analyst\cpa\dbconnect.py", line 64, in fn
    return f(db, *args, **kwargs)
  File "I:\GitHub\CellProfiler-Analyst\cpa\dbconnect.py", line 529, in execute
    '\nSecond exception was: %s'%(connID, query, e, e2))
cpa.dbconnect.DBException: ERROR: Database query failed for connection "MainThread" and failed to reconnect
Query was: "SELECT * FROM Batch1_QC_Per_Image_objecttable LIMIT 1"
First exception was: no such table: Batch1_QC_Per_Image_objecttable
Second exception was: ERROR: Database query failed for connection "MainThread"
Query was: "SELECT * FROM Batch1_QC_Per_Image_objecttable LIMIT 1"
Exception was: no such table: Batch1_QC_Per_Image_objecttable
```

**Comment by jhung0** [comment]
Hopefully fixed with ecda596

**Comment by jhung0** [comment]
@bethac07 can you check?
